client/src/components/ClientProfile/Information/Information.js

```diff
@@ -16,96 +13,190 @@ import VisibilityIcon from '@material-ui/icons/Visibility';
+    const { getCustomerInfo } = props;
     const {
-      email, password, firstName,
-      lastName, phone, address, birthdate,
+      firstName, lastName, telephone,
+      email, oldPassword, newPassword, address, birthdate,
       gender, haveChildren, haveCar, eBicycle,
       checkedSubscribe,
-    } = values;
-    // TODO: complete validation
-    // console.log('%c⧭ values', 'color: #00ff73', values);
-  };
+    } = state;
+    const updatedCustomer = {
+      firstName,
+      lastName,
+      telephone,
+      email,
+      address,
+      birthdate,
+      gender,
+      haveChildren,
+      haveCar,
+      eBicycle,
+      checkedSubscribe,
+    };
 
-  const handleClickShowPassword = () => {
-    setValues({ ...values, showPassword: !values.showPassword });
-  };
-  const handleMouseDownPassword = (event) => {
-    event.preventDefault();
-  };
+    const passwords = {
+      password: oldPassword,
+      newPassword,
+    };
 
-  // const updatedCustomer = {
-  //   firstName: "Antonio",
-  //   lastName: "Front End Developer"
-  // };
+    Object.keys(passwords).forEach((key) => (passwords[key] == null || passwords[key] === '') && delete passwords[key]);
+    if (passwords.newPassword && passwords.password) {
+      axios
+        .put(UPDATE_PASSWORD, passwords)
+        // eslint-disable-next-line no-shadow
+        .then((updatedCustomer) => {
+          successSwalSaveContactInfo();
+          getCustomerInfo();
+        })
+        .catch((err) => {
+          const { data } = err.response;
+          const message = JSON.stringify(data).replace(/{"/g, ' ').replace(/}/g, ' ').replace(/":/g, ':')
+            .replace(/,"/g, '; ');
+          errorSwalSaveContactInfo(message);
+        });
+    }
 
-  // const showItToConsole = () => {
-  //   axios.put('/customers', updatedCustomer)
-  //     // eslint-disable-next-line no-shadow
-  //     .then((updatedCustomer) => {
-  //       console.log('%c⧭ updatedCustomer', 'color: #00e600', updatedCustomer);
-  //     })
-  //     .catch((err) => {
-  //       console.log('%c⧭ err', 'color: #00a3cc', err);
-  //     });
-  // };
+    const reagexpTemplate = '[0-9]{9}';
+    // eslint-disable-next-line max-len
+    if (reagexpTemplate.toLocaleString(updatedCustomer.telephone) && updatedCustomer.telephone.length === 9) {
+      updatedCustomer.telephone = `+380${updatedCustomer.telephone}`;
+    }
 
+    // eslint-disable-next-line max-len
+    Object.keys(updatedCustomer).forEach((key) => (updatedCustomer[key] == null || updatedCustomer[key] === '') && delete updatedCustomer[key]);
+
+    axios
+      .put(UPDATE_CUSTOMER, updatedCustomer)
+      // eslint-disable-next-line no-shadow
+      .then((updatedCustomer) => {
+        successSwalSaveContactInfo();
+        getCustomerInfo();
+      })
+      .catch((err) => {
+        const { data } = err.response;
+        const message = JSON.stringify(data).replace(/{"/g, ' ').replace(/}/g, ' ').replace(/":/g, ':')
```

Comment:
Again, as I said above.

Reply:
The same as I replied in above mentioned statement.